core/glrender/src/main/java/edu/cmu/cs/dennisc/render/gl/imp/adapters/GlrBox.java

```diff
@@ -55,184 +55,148 @@
 import edu.cmu.cs.dennisc.render.gl.imp.RenderContext;
 import edu.cmu.cs.dennisc.scenegraph.Box;
 
+import java.util.HashMap;
+import java.util.Map;
+
 /**
  * @author Dennis Cosgrove
- * Texture is defined by glTexCoord2f, which uses two value to crop the original picture
+ * Texture is defined by glTexCoord2f, which uses two values to crop the original picture
  */
 public class GlrBox extends GlrShape<Box> {
   private void glBox(Context c, boolean isLightingEnabled, boolean isSubElementRequired, boolean isTexturingEnabled) {
     //todo: revist vertex ordering
-    //xMin face
-    //c.gl.glColor3d( 1,1,1 );
     int id = 0;
+    int numberOfCorner = 4;
+    float[][] leftTexCoord = {{.75f, .66f}, {.75f, .33f}, {.5f, .33f}, {.5f, .66f}};
+    float[][] rightTexCoord = {{.25f, .66f}, {.25f, .33f}, {.0f, .33f}, {.0f, .66f}};
+    float[][] bottomTexCoord = {{.5f, .66f}, {.25f, .66f}, {.25f, 1.0f}, {.5f, 1.0f}};
+    float[][] topTexCoord = {{.5f, .0f}, {.25f, .0f}, {.25f, .33f}, {.5f, .33f}};
+    float[][] frontTexCoord = {{.5f, .33f}, {.25f, .33f}, {.25f, .66f}, {.5f, .66f}};
+    float[][] backTexCoord = {{.75f, .66f}, {1.0f, .66f}, {1.0f, .33f}, {.75f, .33f}};
+    double[][] leftVertex = {{this.xMin, this.yMin, this.zMax}, {this.xMin, this.yMax, this.zMax}, {this.xMin, this.yMax, this.zMin}, {this.xMin, this.yMin, this.zMin}};
```

Comment:
Consider renaming to leftFace, since these are the faces of the box, built out of four vertices.